### PR TITLE
Tear down tests properly

### PR DESCRIPTION
**Pull request by simoneduca**
This contains work necessary to [get the mocha `--watch` flag working](https://github.com/zooniverse/Panoptes-Front-End/pull/4233).

Describe your changes.
Clean up spies and stubs after the first test run.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/collections/collaborators.spec.js [hunk 1]
```diff
@@ -51,13 +51,18 @@ describe('<CollectionCollaborators />', function() {
   let wrapper;
   let addUserSpy;
   let deleteUserSpy;
-
-  before(function() {
+  let stub;
```
Comment (eatyourgreens) [inline app/collections/collaborators.spec.js]: Can this use a variable name that describes the stub?

**Comment by eatyourgreens** [comment]
Looks good to me. Can that `stub` variable be more descriptive, and could you edit the commit message to describe what this PR changes?

**Comment by eatyourgreens** [comment]
Is this ready to merge now?

**Comment by simoneduca** [comment]
No sorry, there's one missing test breakage that I'm trying to figure out.

**Comment by simoneduca** [comment]
This is ready to review now.

**Pull request merged**